src/tz_ohos.rs

```diff
@@ -0,0 +1,48 @@
+//! OpenHarmony doesn't have `/etc/localtime`, we have to use it's "Time Service" to get the timezone information:
+//!
+//! - [API Refence](https://gitee.com/openharmony/docs/blob/43726785b4033887cd1a838aaaca5e255897a71e/en/application-dev/reference/apis-basic-services-kit/_time_service.md#oh_timeservice_gettimezone)
+
+use crate::GetTimezoneError;
+use std::ffi::CStr;
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+#[repr(C)]
+#[allow(non_camel_case_types)]
+#[allow(dead_code)]
+enum TimeService_ErrCode {
+    TIMESERVICE_ERR_OK = 0,
+    TIMESERVICE_ERR_INTERNAL_ERROR = 13000001,
+    TIMESERVICE_ERR_INVALID_PARAMETER = 13000002,
+}
+
+#[link(name = "time_service_ndk", kind = "dylib")]
+extern "C" {
+    fn OH_TimeService_GetTimeZone(timeZone: *mut std::ffi::c_char, len: u32)
+        -> TimeService_ErrCode;
+}
+
+/// TODO: Change this `CStr::from_bytes_until_nul` once MSRV was bumped above 1.72.0
+fn from_bytes_until_nul(bytes: &[u8]) -> Option<&CStr> {
+    let nul_pos = bytes.iter().position(|&b| b == 0)?;
+    // SAFETY:
+    // 1. nul_pos + 1 <= bytes.len()
+    // 2. We know there is a nul byte at nul_pos, so this slice (ending at the nul byte) is a well-formed C string.
+    Some(unsafe { CStr::from_bytes_with_nul_unchecked(&bytes[0..=nul_pos]) })
+}
+
+pub(crate) fn get_timezone_inner() -> Result<String, GetTimezoneError> {
+    let mut time_zone = [0; 41];
+    // SAFETY:
+    // `time_zone` is a valid buffer with a length of 40 bytes.
+    let ret =
+        unsafe { OH_TimeService_GetTimeZone(time_zone.as_mut_ptr(), time_zone.len() as u32 - 1) };
+    if ret != TimeService_ErrCode::TIMESERVICE_ERR_OK {
+        return Err(GetTimezoneError::OsError);
+    }
+    let Some(tz_string) =
+        from_bytes_until_nul(&time_zone).map(|x| x.to_string_lossy().into_owned())
+    else {
+        return Err(GetTimezoneError::OsError);
+    };
```

Comment:
[let-else is stabilized in Rust 1.65.0](https://doc.rust-lang.org/rust-by-example/flow_control/let_else.html) which is higher than this crate's MSRV.

```suggestion
    let tz_string = from_bytes_until_nul(&time_zone).ok_or(GetTimezoneError::OsError)?;
    let tz_string = tz_string.to_str().map_err(|_|  GetTimezoneError::OsError)?.to_owned();
```